packages/cozy-dataproxy-lib/src/dataproxy/DataProxyProvider.jsx

```diff
@@ -117,21 +118,59 @@ export const DataProxyProvider = React.memo(({ children, options = {} }) => {
     setDataProxy(() => remote)
   }
 
-  const search = async search => {
-    log.log('Send search query to DataProxy')
-    const result = await dataProxy.search(search, options)
-
-    return result
+  const onReceiveMessage = event => {
+    if (!event.origin.includes('dataproxy')) {
+      return
+    }
+    const eventData = event?.data
+    if (eventData && typeof eventData === 'object') {
+      if (
+        eventData.type === 'DATAPROXYMESSAGE' &&
+        eventData.payload === 'READY'
+      ) {
+        onIframeLoaded()
+      }
+    }
   }
 
-  const value = {
-    dataProxyServicesAvailable,
-    search
-  }
+  useEffect(function () {
+    window.addEventListener('message', onReceiveMessage)
+    return function () {
+      window.removeEventListener('message', onReceiveMessage)
+    }
+  })
```

Comment:
This useEffect does not have any dependency array, this means that it will (un)register the eventListener on every render.

Seems like you'll have to create useCallbacks for onReceiveMessage and onIframeLoaded but those should be pretty stable as the only dependency is on `setDataProxy`

Reply:
Good point, I added the useCallback, does it look ok?